src/com/company/Main.java

```diff
@@ -1,8 +1,270 @@
+            }
+        }
+
+        for (int i = 0; i < checking.length; i++) {
+            if (checking[i][0] == columnRowCount) {
+                System.out.println(WINNER_X + ROW + (i + 1));
+                return false;
+            }
+            if (checking[i][1] == columnRowCount) {
+                System.out.println(WINNER_O + ROW + (i + 1));
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static int getPlayingFieldSize(int max) {
+        System.out.println(PROVIDE_BOARD_SIZE_TEXT);
+        return getNumberFromUser(max);
+    }
+
+    public static int getRowNumber(int max) {
+        System.out.println(PROVIDE_ROW_NUMBER);
+        return getNumberFromUser(max);
+    }
+
+    public static int getNumberFromUser(int max) {
+        int result = 0;
+        do {
+            result = readNumber();
+            if (result < MIN_BOARD_ID || result > max) {
+                System.out.println(WRONG_BOARD_SIZE);
+            }
+            // obsługa esc
+        } while (result < MIN_BOARD_ID || result > max);
+        return result;
+    }
+
+    public static int readNumber() {
+        int result = MIN_BOARD_SIZE - 1;
+        try {
+            result = Integer.parseInt(INPUT.nextLine());
+        } catch (NumberFormatException error) {
+            System.out.println(WRONG_BOARD_SIZE);
+        } catch (Exception error) {
+            error.printStackTrace();
+        }
+        return result;
+    }
+
+    public static int getColumnNumber(int max) {
+        int result = 0;
+        System.out.println(PROVIDE_COLUMN);
+        do {
+            result = readSign() - 'A' + 1;
+            if (result < MIN_BOARD_ID || result > max) {
+                System.out.println(WRONG_BOARD_SIZE);
+            }
+            // obsługa esc
+        } while (result < MIN_BOARD_ID || result > max);
+        return result;
+
+    }
+
+    public static char readSign() {
+        String inputText = INPUT.nextLine();
+        if (inputText.length() != 1) {
+            return BAD_SIGN;
+        }
+        return inputText.toUpperCase().charAt(0);
+    }
+
+    public static void drawField(int rowCount, int[][] tab2) {
+
+        System.out.println("");
+        String firstLine = "     ";     // 3 spacje na początku na kolumnę numerów wierszy
+
+        for (int i = 0; i < rowCount; i++) firstLine += (char) ('a' + i) + " ! ";
+        System.out.println(firstLine);
+
+        String horizontalFullLine = "   ";
+        for (int i = 0; i < rowCount * 4 + 1; i++) horizontalFullLine += "-";
+
+        for (int i = 0; i < rowCount; i++) {
+            System.out.println(horizontalFullLine);
+            // linia z danymi z tabeli
+            String lineWithData = " " + (i + 1) + " |";
+            for (int j = 0; j < rowCount; j++) {
+                switch (tab2[i][j]) {
+                    case ItIsX:
+                        lineWithData += " X |";
```

Comment:
Myślałaś, żeby wyrzucić ` |"` na koniec funkcji i dodawać zawsze? A najlepiej zrobić formatowany String `%s |"` i wrzucać tam poprostu odpowiedni znak?
To też nam pomoże jak dojdziemy do lekcji o Enumach.